packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ExtractionService.java

```
@@ -360,145 +362,81 @@ public void run() {
+					 if (document.getText() == null || document.getText().isEmpty()) {
+						 throw new RuntimeException("No text found in paper document");
+					 }
+
+					 // add optional models
+					 final List<Model> models = new ArrayList<>();
+					 for (final UUID modelId : modelIds) {
+						 models.add(modelService.getAsset(modelId).orElseThrow());
+					 }
+					 clientInterface.sendMessage("Model(s) found, added to extraction request.");
+
+					 // Create a collection to hold the variable extractions
+					 JsonNode collection = null;
+
+					 clientInterface.sendMessage("Sending request to be processes by SKEMA and MIT.");
+					 final IntegratedTextExtractionsBody body = new IntegratedTextExtractionsBody(document.getText(), models);
+					 final ResponseEntity<JsonNode> resp = skemaUnifiedProxy.integratedTextExtractions(body);
+
+					 clientInterface.sendMessage("Response received.");
+					 if (resp.getStatusCode().is2xxSuccessful()) {
+						 for (final JsonNode output : resp.getBody().get("outputs")) {
+							 if (!output.has("errors") || output.get("errors").isEmpty()) {
+								 collection = output.get("data");
+								 break;
+							 }
+						 }
+					 } else {
+						 throw new RuntimeException("non successful response.");
+					 }
+
+					 if (collection == null) {
+						 throw new RuntimeException("No variables extractions returned");
+					 }
+
+					 clientInterface.sendMessage("Organizing and saving the extractions.");
+					 final List<JsonNode> attributes = new ArrayList<>();
+					 for (final JsonNode attribute : collection.get("attributes")) {
+						 attributes.add(attribute);
+					 }
+
+					 // add the attributes to the metadata
+					 if (document.getMetadata() == null) {
+						 document.setMetadata(new HashMap<>());
+					 }
+					 document.getMetadata().put("attributes", attributes);
+
+					 // update the document
+					 documentService.updateAsset(document).orElseThrow();
+
+				 } catch (final Exception e) {
```

This will eat any exceptions thrown by the proxied services (skema unified, etc). Consider adding something like:

```
catch (final FeignException e) {
					final String error = "Transitive service failure";
					throw new ResponseStatusException(
							e.status()<100? HttpStatus.INTERNAL_SERVER_ERROR: HttpStatus.valueOf(e.status()),
							error + ": " + e.getMessage());
				}
```